server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/SieveScriptRoutesTest.java

```diff
@@ -0,0 +1,267 @@
+import org.apache.james.sieverepository.api.ScriptContent;
+import org.apache.james.sieverepository.api.ScriptName;
+import org.apache.james.sieverepository.api.SieveRepository;
+import org.apache.james.sieverepository.api.exception.ScriptNotFoundException;
+import org.apache.james.sieverepository.api.exception.StorageException;
+import org.apache.james.sieverepository.file.SieveFileRepository;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.user.memory.MemoryUsersRepository;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+
+public class SieveScriptRoutesTest {
+
+    private static final String SIEVE_ROOT = FileSystem.FILE_PROTOCOL + "sieve";
+
+    private FileSystem fileSystem;
+
+    private WebAdminServer webAdminServer;
+    private SieveRepository sieveRepository;
+    private UsersRepository usersRepository;
+    private String sieveContent;
+
+    @BeforeEach
+    public void setUp() throws ConfigurationException, IOException, UsersRepositoryException {
+        this.fileSystem = new FileSystem() {
+            @Override
+            public File getBasedir() throws FileNotFoundException {
+                return new File(System.getProperty("java.io.tmpdir"));
+            }
+
+            @Override
+            public InputStream getResource(String url) throws IOException {
+                return new FileInputStream(getFile(url));
+            }
+
+            @Override
+            public File getFile(String fileURL) throws FileNotFoundException {
+                return new File(getBasedir(), fileURL.substring(FileSystem.FILE_PROTOCOL.length()));
+            }
+        };
+
+        sieveRepository = new SieveFileRepository(fileSystem);
+        usersRepository = MemoryUsersRepository.withoutVirtualHosting();
+        usersRepository.addUser("userA", "password");
+
+        URL sieveResource = ClassLoader.getSystemResource("sieve/my_sieve");
+        sieveContent = IOUtils.toString(sieveResource, StandardCharsets.UTF_8);
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new SieveScriptRoutes(sieveRepository, usersRepository));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .build();
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        File root = fileSystem.getFile(SIEVE_ROOT);
+        // Remove files from the previous test, if any
+        if (root.exists()) {
+            FileUtils.forceDelete(root);
```

Comment:
there's a @TestRule for that.
Anyway, I don't see why we should use files for that

Reply:
`SieveFileRepository` l 91

Reply:
How about I using `FileSystem` with `TemporaryFolderExtension`

Reply:
> SieveFileRepository l 91

Well, it's not an explanation. Why using a file implementation at all ?